Track test counts per execution.

WangyangZhou90 opened the pull request:
Previous code [Tests ran counter by VincentBu · Pull Request #111145 · dotnet/runtime](https://github.com/dotnet/runtime/pull/111145) added test run counter which records how many times each test run and write them at the end of logging file. However it doesn't work when tests is killed by assert failures or other errors. This pr aims to record each time a test is completed.
the log before looks like:
```
[5/14/2025 8:34:28 PM 1] RF.StartTest, RTs(7) - Instances of this test: 2 - New Test:GCPerfSim - LOH A Lot Of Live Data., 31 threads
[5/14/2025 8:34:28 PM 1] RUN 5/14/2025 8:34:28 PM Test Started: GCPerfSim - POH Scenario.
 3
[5/14/2025 8:34:28 PM 1] RF.StartTest, RTs(8) - Instances of this test: 2 - New Test:GCPerfSim - POH Scenario., 32 threads
[5/14/2025 8:34:28 PM 1] RUN 5/14/2025 8:34:28 PM Test Started: GCPerfSim - LOH No Live Data.
 0
[5/14/2025 8:34:28 PM 1] RF.StartTest, RTs(9) - Instances of this test: 3 - New Test:GCPerfSim - LOH No Live Data., 33 threads
[5/14/2025 8:34:28 PM 1] RUN 5/14/2025 8:34:28 PM Test Started: GCPerfSim - LOH Some Live Data.
 1
[5/14/2025 8:34:28 PM 1] RF.StartTest, RTs(10) - Instances of this test: 3 - New Test:GCPerfSim - LOH Some Live Data., 34 threads
[5/14/2025 8:34:28 PM 1] RUN 5/14/2025 8:34:28 PM Test Started: GCPerfSim - LOH A Lot Of Live Data.
 2
[5/14/2025 8:34:28 PM 1] RF.StartTest, RTs(11) - Instances of this test: 3 - New Test:GCPerfSim - LOH A Lot Of Live Data., 35 threads
[5/14/2025 8:34:28 PM 1] RUN 5/14/2025 8:34:28 PM Test Started: GCPerfSim - POH Scenario.
```
after:
```
[5/20/2025 2:20:32 PM 1] RF.StartTest, RTs(7) - Instances of this test: 2 - New Test:GCPerfSim - LOH A Lot Of Live Data., 31 threads
[5/20/2025 2:20:32 PM 1] Run tests "GCPerfSim - LOH A Lot Of Live Data." 2 times.
[5/20/2025 2:20:32 PM 1] RUN 5/20/2025 2:20:32 PM Test Started: GCPerfSim - POH Scenario.
 3
[5/20/2025 2:20:32 PM 1] RF.StartTest, RTs(8) - Instances of this test: 2 - New Test:GCPerfSim - POH Scenario., 32 threads
[5/20/2025 2:20:32 PM 1] Run tests "GCPerfSim - POH Scenario." 2 times.
[5/20/2025 2:20:32 PM 1] RUN 5/20/2025 2:20:32 PM Test Started: GCPerfSim - LOH No Live Data.
 0
[5/20/2025 2:20:32 PM 1] RF.StartTest, RTs(9) - Instances of this test: 3 - New Test:GCPerfSim - LOH No Live Data., 33 threads
[5/20/2025 2:20:32 PM 1] Run tests "GCPerfSim - LOH No Live Data." 3 times.
[5/20/2025 2:20:32 PM 1] RUN 5/20/2025 2:20:32 PM Test Started: GCPerfSim - LOH Some Live Data.
 1
[5/20/2025 2:20:32 PM 1] RF.StartTest, RTs(10) - Instances of this test: 3 - New Test:GCPerfSim - LOH Some Live Data., 34 threads
[5/20/2025 2:20:32 PM 1] Run tests "GCPerfSim - LOH Some Live Data." 3 times.
[5/20/2025 2:20:32 PM 1] RUN 5/20/2025 2:20:32 PM Test Started: GCPerfSim - LOH A Lot Of Live Data.
 2
[5/20/2025 2:20:32 PM 1] RF.StartTest, RTs(11) - Instances of this test: 3 - New Test:GCPerfSim - LOH A Lot Of Live Data., 35 threads
[5/20/2025 2:20:32 PM 1] Run tests "GCPerfSim - LOH A Lot Of Live Data." 3 times.
[5/20/2025 2:20:32 PM 1] RUN 5/20/2025 2:20:32 PM Test Started: GCPerfSim - POH Scenario.
 3
[5/20/2025 2:20:32 PM 1] RF.StartTest, RTs(12) - Instances of this test: 3 - New Test:GCPerfSim - POH Scenario., 36 threads
[5/20/2025 2:20:32 PM 1] Run tests "GCPerfSim - POH Scenario." 3 times.
```

WangyangZhou60 committed:
add test count after tests run

Maoni0 commented:
Copilot had good suggestions. 

also please add some description to the PR - is there already per test logging? if so what value does this add (was the current logging insufficient)? what does the log look like before and after your change? and etc.

WangyangZhou60 committed:
Merge branch 'dotnet:main' into TestRunCounter

WangyangZhou60 committed:
Merge branch 'dotnet:main' into TestRunCounter

WangyangZhou60 committed:
Merge branch 'dotnet:main' into TestRunCounter

WangyangZhou60 committed:
PR feedback

WangyangZhou60 committed:
Delete the redundant Spaces

WangyangZhou90 commented:
> Copilot had good suggestions.
> 
> also please add some description to the PR - is there already per test logging? if so what value does this add (was the current logging insufficient)? what does the log look like before and after your change? and etc.

thanks for the feedback, I've updated the code and added some description to the PR.

mrsharm reviewed (approved):
LGTM

shimingsg committed:
Merge branch 'main' into TestRunCounter

shimingsg committed:
Merge branch 'main' into TestRunCounter

shimingsg committed:
Merge branch 'main' into TestRunCounter
